projects/vdk-core/tests/functional/run/test_run_standalone_data_job.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from pathlib import Path
+
+from functional.run import util
+from vdk.internal.builtin_plugins.run.standalone_data_job import (
+    StandaloneDataJobFactory,
+)
+
+
+def test_standalone_data_job():
+    data_job_directory = Path(util.job_path("simple-job"))
+
+    with StandaloneDataJobFactory.create(data_job_directory) as job_input:
+        assert job_input.get_name() == "simple-job"
```

Comment:
@tozka I'd like to add a bunch of tests that validate that all the correct vdk hooks have been called in the right order.

Is there an example of a test already doing this?  Thx

Reply:
I don't think we have a test just for the order. That seems like an omission that we have not realized until now. 
Certainly, some tests would fail if the order is not kept (e.g vdk_initialize hook often relies on some configuration from vdk_configure hook). 

But not all - I switched the order of execution of "vdk_command_line" and "vdk_configure" in the code and re-run all tests and nothing failed !!! (Something should have failed). 

I can suggest implementing a few hooks and checking the order.
The order is documented at https://github.com/vmware/versatile-data-kit/tree/main/projects/vdk-plugins#types-of-plugins 
Something like : 
```
class OrderedHooks:
    calls = []

    @hookimpl
    def vdk_configure(...):
        self.calls.append("vdk_configure")

 ....

```


CliEntryBasedTestRunner might be helpful also: https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-test-utils/src/vdk/plugin/test_utils/util_funcs.py#L81

Reply:
CliEntryBasedTestRunner may not be as helpful for your case. I guess you can inject PluginRegistry and use             plugin_registry.load_plugin_with_hooks_impl(plugin) to load your testing plugins

Reply:
Thanks for the pointers.  [517e334](https://github.com/vmware/versatile-data-kit/pull/793/commits/517e3349838f0c6e245cd1856c5b76e68ab47018) contains my first attempt at these kinds of tests